### ISSUE TITLE
Question about cath_label_mapping.pt

### ISSUE THREAD

**Issue by junhaobearxiong** [NVIDIA-Digital-Bio/proteina#4]
Hi!

Thank you for sharing this really cool work! I have a question about the file `cath_label_mapping.pt` that maps from CATH code to an index (integer) used by the networks. 

In both Appendix G.3.1 in the paper and the architecture of the fold classifier, it seems like the output logit size are 5 for the C-level, 43 for the A-level and 1336 for the T-level (e.g. [here](https://github.com/NVIDIA-Digital-Bio/proteina/blob/15e5b5e2935a5f4d81e32dd4987a903eef352bcb/proteinfoundation/metrics/gearnet_utils.py#L460)). However, in the `cath_label_mapping.pt`, it seems like there are 1472 unique labels for the T-level. I just want to confirm how the 1336 output dimensions for the T-level of the fold classifier correspond to the 1472 unique labels in `cath_label_mapping.pt`? 

Perhaps I overlooked something, but would appreciate some pointers here :)

Thanks so much!

Best,
Bear

**Comment by Oxer11** [comment]
Thanks for the question and digging deeply into our codebase.  

The file `cath_label_mapping.pt` here is used for the fold embedding in [our model feature factory](https://github.com/NVIDIA-Digital-Bio/proteina/blob/15e5b5e2935a5f4d81e32dd4987a903eef352bcb/proteinfoundation/nn/feature_factory.py#L176). For legacy issue, the cath code mapping used in the fold classifier is different from the fold embedding used in the model. The main reason for this is when training the fold classifier, we did filtering for the PDB dataset, which leads to fewer T-level labels (1,336 after filtering v.s. 1,472 before filtering). In contrast, the Proteina models are trained on much larger dataset, for which we need to include all unique fold labels.

In short, the `cath_label_mapping.pt` cannot be directly used for interpreting the output logits of the fold classifier. We'll think about how to share the cath label mapping for the fold classifier.

**Comment by Eikor** [comment]
Is there any update on the cath label mapping for the fold classifier? 

Thank you!
